Stratis.Bitcoin.Features.Wallet/WalletSyncManager.cs

```diff
@@ -85,22 +98,25 @@ public virtual void ProcessBlock(Block block)
                     // The current wallet hash was not found on the main chain.
                     // A reorg happenend so bring the wallet back top the last known fork.
 
-                    var fork = this.walletTip;
+                    var fork = walletTip;
 
                     // We walk back the chained block object to find the fork.
                     while (this.chain.GetBlock(fork.HashBlock) == null)
                         fork = fork.Previous;
 
                     this.walletManager.RemoveBlocks(fork);
+
+                    // update temporary wallet tip to be able to catch up again.
+                    walletTip = this.chain.GetBlock(fork.HashBlock);
```

Comment:
Ah I see why I missed this, previously the second if was an else so I didn't need to remember the fork.

Good catch.